### PR TITLE
fix(functions): inject default CORS headers

### PR DESCRIPTION
**Pull request by dbm03**
### **PR Type**
Enhancement

___

### **Description**
- Inject default `Access-Control-Allow-Origin: *` and `Access-Control-Allow-Headers: origin,Accept,Authorization,Content-Type` on every function response, only when the user function hasn't already set them. Implementation wraps `res.writeHead` so the check runs after user code (and middleware like the `cors` package) has had a chance to set its own values.
- Allow user code to opt out of the defaults by setting either header to an empty string.
- Rewrite the CORS guide in the docs to reflect the new default behavior, document the disable-by-empty-string pattern, the credentials caveat, and the fact that the runtime does not respond to `OPTIONS` on the user's behalf.
- Add three integration tests covering default presence, user override, and the empty-string disable path; add `cors-custom.js` and `cors-disabled.js` example fixtures across all three example projects (npm/pnpm/yarn).
- Add `functions` as a valid `PKG` token in the PR title format documented in `PULL_REQUEST_TEMPLATE.md`.

___

### Diagram Walkthrough

```mermaid
flowchart LR
  Req["HTTP request"] --> CorsMW["CORS default middleware<br/>(wraps res.writeHead)"]
  CorsMW --> Body["express.json / urlencoded"]
  Body --> UserFn["User function<br/>(may setHeader / use cors pkg)"]
  UserFn --> WriteHead["res.writeHead patched"]
  WriteHead -->|"hasHeader(ACAO)?"| Defaults{"Already set?"}
  Defaults -- "no" --> Inject["Inject *<br/>and default ACAH"]
  Defaults -- "yes" --> Skip["Keep user value"]
  Inject --> Out["Response to client"]
  Skip --> Out
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody>
<tr><td><strong>Core logic</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>local-wrapper.js</strong><dd><code>Wrap res.writeHead to inject default CORS headers when the user hasn't set them</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-d76e5132bd5d3080a242c743342e5dcc91457ae322af2c5607a20305b4bfaa6c">+21/-0</a></td>
</tr>
</table></details></td></tr>
<tr><td><strong>Tests</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>integration.test.js</strong><dd><code>Cover default-present, override, and empty-string disable cases; register new example routes</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-ceb6cbd64208432d103ee0ab22065cd63b62742f974fb2d61be2df269adc6094">+29/-0</a></td>
</tr>
</table></details></td></tr>
<tr><td><strong>Example fixtures</strong></td><td><details><summary>6 files</summary><table>
<tr>
  <td><strong>example-npm/functions/cors-custom.js</strong><dd><code>Sample function overriding default CORS headers</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-2028cbd2fe1d1e2b2c1342e2f1ab432fbcc4a92ae60631d655902c2fc2bacd98">+5/-0</a></td>
</tr>
<tr>
  <td><strong>example-npm/functions/cors-disabled.js</strong><dd><code>Sample function disabling default CORS via empty strings</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-b25deab4b154942c190cc35102a337886712af7e88950c0c9500c8db7a2602d1">+5/-0</a></td>
</tr>
<tr>
  <td><strong>example-pnpm/functions/cors-custom.js</strong><dd><code>Same override sample for the pnpm example project</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-e1bcebae9cb768cf35e3d615f18850608e8a2ba9eefbde7a5fbda1ae42eda099">+5/-0</a></td>
</tr>
<tr>
  <td><strong>example-pnpm/functions/cors-disabled.js</strong><dd><code>Same disable sample for the pnpm example project</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-ce065ce840407bd3f416a8144498cf1ca7fa286846f79e9d3b51b9534a1428b4">+5/-0</a></td>
</tr>
<tr>
  <td><strong>example-yarn/functions/cors-custom.js</strong><dd><code>Same override sample for the yarn example project</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-fa30ebd8664a93b336b67e64eceab0bc4788c1c8a0eadd03a3a9e3919ec6648e">+5/-0</a></td>
</tr>
<tr>
  <td><strong>example-yarn/functions/cors-disabled.js</strong><dd><code>Same disable sample for the yarn example project</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-6090c869e27316cac2183e580fca960e7b5ac40dadd6be77dd73024ef9d78208">+5/-0</a></td>
</tr>
</table></details></td></tr>
<tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>cors.mdx</strong><dd><code>Rewrite the CORS guide to describe defaults, disable, customize, credentials, and preflight</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-951770b10d7713649e0d36338fc18c9bf1015b08978b69eb3f92c437dcd7dc2d">+58/-31</a></td>
</tr>
</table></details></td></tr>
<tr><td><strong>Configuration</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>PULL_REQUEST_TEMPLATE.md</strong><dd><code>Add `functions` to the list of valid PKG tokens for PR titles</code></dd></td>
  <td><a href="https://github.com/nhost/nhost/pull/4177/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35">+1/-0</a></td>
</tr>
</table></details></td></tr>
</tbody></table>

</details>

___

### CODE REVIEW

File: docs/src/content/docs/products/functions/guides/cors.mdx [hunk 1]
```diff
@@ -4,11 +4,26 @@ description: Handle Cross-Origin Resource Sharing in Nhost Functions
 keywords: ["CORS", "cross-origin", "serverless functions", "Access-Control", "preflight", "OPTIONS"]
 ---
 
-Nhost Functions do not set any CORS headers automatically. If your function is called from a browser on a different origin (for example, a frontend app on `localhost:5173` calling functions at `local.functions.local.nhost.run`), you need to handle CORS yourself.
+Nhost Functions set a small set of default CORS headers so simple cross-origin `GET` and `POST` requests work without extra code. Other methods, custom headers, or credentials are your responsibility.
```
Comment (dbarrosop) [inline docs/src/content/docs/products/functions/guides/cors.mdx]: I think we need to make this a bit clearer to indicate that by default we set X headers unless the function itself does it and to clarify if you don't want us to set any CORS header  you can return an empty `Access-Control-Allow-Origin`

File: services/functions/test/integration.test.js [hunk 2]
```diff
@@ -58,6 +58,15 @@ describe.each([
     expect(await res.text()).toContain('Hello, World!');
   });
 
+  it('default CORS headers are present on user function responses', async () => {
```
Comment (dbarrosop) [inline services/functions/test/integration.test.js:63]: can you add a test with a function disabling the cors header by returning it empty and another one by setting their own?

File: docs/src/content/docs/products/functions/guides/cors.mdx (hunk 1 above)
Comment (dbm03) [reply docs/src/content/docs/products/functions/guides/cors.mdx]: I'm adding this:

Nhost Functions inject a small set of default CORS headers so simple cross-origin `GET` and `POST` requests work without extra code. The defaults only fill in headers your function doesn't set. Anything you set yourself (directly or via middleware like [`cors`](https://www.npmjs.com/package/cors)) is preserved as-is. Other methods, custom headers, or credentials are your responsibility.

## Default Headers

Unless your function sets them itself, every response includes:

| Header                          | Value                                          |
| ------------------------------- | ---------------------------------------------- |
| `Access-Control-Allow-Origin`   | `*`                                            |
| `Access-Control-Allow-Headers`  | `origin,Accept,Authorization,Content-Type`     |

`Access-Control-Allow-Methods` and `Access-Control-Allow-Credentials` are not set. Requests that need them won't work until you set them yourself.

:::caution
`*` allows any website to call your function. In production, restrict the origin to your actual frontend(s).
:::

## Disabling Default CORS

If you want to block cross-origin access, return an empty `Access-Control-Allow-Origin` and/or `Access-Control-Allow-Headers`:

```ts title="./functions/api.ts"
res.setHeader('Access-Control-Allow-Origin', '')
res.setHeader('Access-Control-Allow-Headers', '')
```

File: services/functions/test/integration.test.js (hunk 2 above)
Comment (dbm03) [reply services/functions/test/integration.test.js:63]: I've just added these in the last commit:
https://github.com/nhost/nhost/blob/8e8c0d4991ae6cb4e7b0be063a55a2355725c690/services/functions/test/integration.test.js#L72-L88

**Review by dbarrosop — APPROVED** [review]
